beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -0,0 +1,159 @@
+package apimiddleware
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prysmaticlabs/prysm/shared/gateway"
+	"github.com/prysmaticlabs/prysm/shared/grpcutils"
+)
+
+type sszConfig struct {
+	sszPath      string
+	fileName     string
+	responseJson sszResponseJson
+}
+
+func handleGetBeaconStateSsz(m *gateway.ApiProxyMiddleware, endpoint gateway.Endpoint, writer http.ResponseWriter, request *http.Request) (handled bool) {
+	config := sszConfig{
+		sszPath:      "/eth/v1/debug/beacon/states/{state_id}/ssz",
+		fileName:     "beacon_state.ssz",
+		responseJson: &beaconStateSszResponseJson{},
+	}
+	return handleGetSsz(m, endpoint, writer, request, config)
+}
+
+func handleGetBlockSsz(m *gateway.ApiProxyMiddleware, endpoint gateway.Endpoint, writer http.ResponseWriter, request *http.Request) (handled bool) {
+	config := sszConfig{
+		sszPath:      "/eth/v1/beacon/blocks/{block_id}/ssz",
+		fileName:     "beacon_block.ssz",
+		responseJson: &blockSszResponseJson{},
+	}
+	return handleGetSsz(m, endpoint, writer, request, config)
+}
+
+func handleGetSsz(
+	m *gateway.ApiProxyMiddleware,
+	endpoint gateway.Endpoint,
+	writer http.ResponseWriter,
+	request *http.Request,
+	config sszConfig,
+) (handled bool) {
+	if !sszRequested(request) {
+		return false
+	}
+
+	if errJson := prepareSszRequestForProxying(m, endpoint, request, config.sszPath); errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	grpcResponse, errJson := gateway.ProxyRequest(request)
+	if errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	grpcResponseBody, errJson := gateway.ReadGrpcResponseBody(grpcResponse.Body)
+	if errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	if errJson := gateway.DeserializeGrpcResponseBodyIntoErrorJson(endpoint.Err, grpcResponseBody); errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	if endpoint.Err.Msg() != "" {
+		gateway.HandleGrpcResponseError(endpoint.Err, grpcResponse, writer)
+		return true
+	}
+	if errJson := gateway.DeserializeGrpcResponseBodyIntoContainer(grpcResponseBody, config.responseJson); errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	responseSsz, errJson := serializeMiddlewareResponseIntoSsz(config.responseJson.SszData())
+	if errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	if errJson := writeSszResponseHeaderAndBody(grpcResponse, writer, responseSsz, config.fileName); errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+	if errJson := gateway.Cleanup(grpcResponse.Body); errJson != nil {
+		gateway.WriteError(writer, errJson, nil)
+		return true
+	}
+
+	return true
+}
+
+func sszRequested(request *http.Request) bool {
```

Comment:
Consider naming to reflect a boolean return type. I.e. `isSSZRequested`.

Also, I think it's the "go way" to always capitalize acronyms. i.e. SSZ, URL, FTP, etc. Consider renaming all `Ssz` to `SSZ`, within reason